doc: nrf5340audio: add docs for new apps (#37)
alexsven/sdk-nrf: applications/nrf5340_audio/doc/configuration.rst

.. _nrf53_audio_app_configuration:

Configuring the nRF5340 applications
####################################

.. contents::
   :local:
   :depth: 2

|config|

.. _nrf53_audio_app_configuration_select_bidirectional:

Selecting the CIS bidirectional communication
*********************************************

> The CIS unidirectional mode is the default operating mode for the application.

Reviewer: I think you noted this down in the meeting, but this will have to change

To switch to the bidirectional mode, set the :ref:`CONFIG_STREAM_BIDIRECTIONAL <config_nrf53_audio_app_options>` Kconfig option to ``y``  in the :file:`prj.conf` file (for the debug version) or in the :file:`prj_release.conf` file (for the release version).

.. _nrf53_audio_app_configuration_enable_walkie_talkie:

Enabling the walkie-talkie demo
===============================

The walkie-talkie demo uses one or two bidirectional streams from the gateway to one or two headsets.
The PDM microphone is used as input on both the gateway and headset device.
To switch to using the walkie-talkie, set the :ref:`CONFIG_WALKIE_TALKIE_DEMO <config_nrf53_audio_app_options>` Kconfig option to ``y``  in the :file:`prj.conf` file (for the debug version) or in the :file:`prj_release.conf` file (for the release version).

.. _nrf53_audio_app_configuration_select_i2s:

Selecting the I2S serial
************************

In the default configuration, the gateway application uses the USB serial port as the audio source.

> The :ref:`nrf53_audio_app_building` and :ref:`nrf53_audio_app_testing` steps also refer to using the USB serial connection.

Reviewer: I guess these refs are invalid now? Would probably be the same for most of the refs starting with nrf53_audio_app_ in this file

Author: These are still valid, pointing to pages in the doc/ for the app.

.. _nrf53_audio_app_configuration:

> Configuring the nRF5340 applications

Reviewer 2: ```suggestion
Configuring the Audio applications
``` ?

####################################

.. contents::
   :local:
   :depth: 2

|config|

.. _nrf53_audio_app_configuration_select_bidirectional:

Selecting the CIS unidirectional communication

> *********************************************

Reviewer 2: ```suggestion
**********************************************
```

> The CIS bidirectional mode is the default operating mode for the application.

Reviewer: I'm wondering how we should phrase this: There are now four applications, but buildprog will build the CIS ones by default.

Author: Attempted a rephrase.

To switch to the unidirectional mode, set the :ref:`CONFIG_STREAM_BIDIRECTIONAL <config_nrf53_audio_app_options>` Kconfig option to ``n``  in the :file:`prj.conf` file (for the debug version) or in the :file:`prj_release.conf` file (for the release version).

.. _nrf53_audio_app_configuration_enable_walkie_talkie:

Enabling the walkie-talkie demo
===============================

The walkie-talkie demo uses one or two bidirectional streams from the gateway to one or two headsets.
The PDM microphone is used as input on both the gateway and headset device.
To switch to using the walkie-talkie, set the :ref:`CONFIG_WALKIE_TALKIE_DEMO <config_nrf53_audio_app_options>` Kconfig option to ``y``  in the :file:`prj.conf` file (for the debug version) or in the :file:`prj_release.conf` file (for the release version).

.. _nrf53_audio_app_configuration_select_i2s:

> Selecting the I2S serial

Reviewer 2: ```suggestion
Selecting analog jack input using I2S instead of USB
```

Author: Applied, but without "instead of USB".

************************

> In the default configuration, the gateway application uses the USB serial port as the audio source.

Reviewer 2: ```suggestion
In the default configuration, the gateway application uses USB as the audio source.
```

The :ref:`nrf53_audio_app_building` and :ref:`nrf53_audio_app_testing` steps also refer to using the USB serial connection.

> To switch to using the I2S serial connection, set the :ref:`CONFIG_AUDIO_SOURCE_I2S <config_nrf53_audio_app_options>` Kconfig option to ``y`` in the :file:`prj.conf` file for the debug version and in the :file:`prj_release.conf` file for the release version.

Reviewer 2: ```suggestion
To switch to using the 3.5 mm jack analog input, set the :ref:`CONFIG_AUDIO_SOURCE_I2S <config_nrf53_audio_app_options>` Kconfig option to ``y`` in the :file:`prj.conf` file for the debug version and in the :file:`prj_release.conf` file for the release version.
```

When testing the application, an additional audio jack cable is required to use I2S.
Use this cable to connect the audio source (PC) to the analog **LINE IN** on the development kit.

.. _nrf53_audio_app_adding_FEM_support:

Adding FEM support
******************

You can add support for the nRF21540 front-end module (FEM) to the following nRF5340 Audio applications:

* :ref:`Broadcast source <nrf53_audio_broadcast_source_app>`
* :ref:`Unicast client <nrf53_audio_unicast_client_app>`
* :ref:`Unicast server <nrf53_audio_unicast_server_app>`

> The :ref:`Broadcast sink application <nrf53_audio_broadcast_sink_app>` does not support FEM.

Reviewer 2: ```suggestion
The :ref:`Broadcast sink application <nrf53_audio_broadcast_sink_app>` does not support, nor need FEM as it only receives data.
```
